src/main/java/bio/terra/cli/apps/LocalProcessCommandRunner.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.cli.apps;
+
+import bio.terra.cli.apps.utils.LocalProcessLauncher;
+import bio.terra.cli.context.GlobalContext;
+import bio.terra.cli.context.WorkspaceContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** This class runs client-side tools in a local process. */
+public class LocalProcessCommandRunner extends CommandRunner {
+  private static final Logger logger = LoggerFactory.getLogger(LocalProcessCommandRunner.class);
+
+  private LocalProcessLauncher localProcessLauncher = new LocalProcessLauncher();
+
+  public LocalProcessCommandRunner(GlobalContext globalContext, WorkspaceContext workspaceContext) {
+    super(globalContext, workspaceContext);
+  }
+
+  /**
+   * Run a tool command in a local process.
+   *
+   * <p>Some setup/cleanup commands for gcloud authentication and configuration will be run
+   * before/after the given command.
+   *
+   * @param command the command and arguments to execute
+   * @param envVars a mapping of environment variable names to values
+   */
+  protected void runToolCommandImpl(List<String> command, Map<String, String> envVars) {
+    List<String> bashCommands = new ArrayList<>();
+    bashCommands.add("gcloud config set project " + workspaceContext.getGoogleProject());
+    bashCommands.add(
+        "gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS");
```

Comment:
@wchamber For launching apps in a local process, I included commands to set the `gcloud` project and account config properties, in keeping with what we do for the Docker container. However, if the main use case for launching in a local process is when running on a notebook VM, I'm wondering if I should skip this.

<img width="626" alt="Screen Shot 2021-05-14 at 5 32 08 PM" src="https://user-images.githubusercontent.com/12446128/118335640-16038180-b4de-11eb-9f33-272d2a15685e.png">
Seems like the project is already set correctly and I think we're expecting users to do `gcloud auth login` so those credentials should be fine to talk to Terra services. What do you think?

Reply:
Thanks to you both for the comments and discussion here and at stand-up. Here's my brief summary and proposed path forward. Feedback welcome.

- Users running via local process either want to use a tool that is not yet supported (e.g. `dsub`) or they want to run that tool outside of a docker container (e.g. `nextflow` container mode).
- On one hand, it's confusing to have commands run via docker work differently than those run via local process. 
- On the other hand, assumptions for each option are likely different. Local process users have tools setup already on their local machine, and may not want us messing with their configs, while Docker users probably do not.
- I like the suggestion of just setting the project, not the credentials. I'm going to do that for local process and keep the docker as is. I'm not going to include a flag to modify this behavior in the PR, will leave that for a future request/PR.

- I wrote a [doc](https://docs.google.com/document/d/1ruzjYOxsvs2O02NZ_hxzeu2AK9iLMT68T5k_KyxWktA/edit#heading=h.ga24akdcomc1) a few months ago with different approaches we can take here (docker, local process, source env vars). I don't think any of these options are mutually exclusive and all seem like good fits for certain use cases. So regarding our discussion about sourcing a script, that still seems like a good enhancement to me, just orthogonal to the question of how to make the local process work.

Reply:
Sounds good to me.
I think we need to make sure to cleanup the gcloud setting because afaik it's global and not process specific. (Also could suffer from concurrent modification/gcloud execution.)

I'm still not totally sold on having local process support instead of just having docker & source env vars, though I do see how it's analogous to running with docker. But I think we should go ahead and support all 3 to get feedback and see the different use cases.

(Should I do a review pass now, or are there now changes to do first?)

Reply:
Haha glad we're on the same page! I also thought restoring the setting would be good and just requested feedback on it.

I think all of this tool launching is just guessing. I'm not sold on any of the options. Would users rather install (so no need for Docker to bundle versions that work) and configure (so no need for the gcloud setup/cleanup) things themselves, and just use `terra resolve` instead of environment variables? That would certainly be my personal preference, at least given the small number of easy-to-install tools we "support" today (gcloud & nextflow are both simple to install).

I see this as just another option to present to users to see what they like. Specific questions like whether to modify the `gcloud` account configuration seem pretty far down into the weeds to me, and perhaps we should wait for a request from a user.

Reply:
Yes, all set to review, thanks!

Reply:
FWIW (and probably not super helpful), but I’d also personally prefer to install the tools side-by-side with our CLI, and I’d probably do it in a Docker container to keep my system clean and lock down on desired versions.  But I’m guessing that might not be especially representative of our users.  

But publishing a curated docker image like this to DockerHub might be another option?

Reply:
Yes, users could build their own image, and then set the `image` configuration property with the CLI. They would have to copy in the `terra_init.sh` script where the CLI expects it to be, but we could relax that restriction. So that's technically possible now and we could easily make it less cumbersome.

I'm just not sure about the original motivation for using a Docker image in the first place: to provide a set of tools+versions that we know work well together. At the moment, I don't think we have enough tools to justify that. But, I think like most decisions around the CLI, it's a hedge because we currently have no users (or very close to it, since the solutions team is just testing it out, not actually working with it, as I understand).